### PR TITLE
DBZ-191 Mentioning default values in MySQL connector docs

### PR DESCRIPTION
**Pull request by gunnarmorling**
@jpechane @sweat123 Here's a quick update to docs regarding the default value issue.

### CODE REVIEW

File: docs/connectors/mysql.asciidoc [hunk 1]
```diff
@@ -917,6 +917,12 @@ Columns that store strings are defined in MySQL with a character set and collati
 
 Other data type mappings are described in the following sections.
 
+If present, a column's default value will be propagated to the corresponding field's Kafka Connect schema.
```
Comment (jpechane) [inline docs/connectors/mysql.asciidoc]: @gunnarmorling IIRC this is true also for cases where default value is set to all zeroes.

**Comment by gunnarmorling** [comment]
> his is true also for cases where default value is set to all zeroes.

I'm not quite sure what you mean. Isn't this a form of "If present"?

**Comment by jpechane** [comment]
@gunnarmorling Checkt this https://github.com/debezium/debezium/blob/master/debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlDefaultValuePreConverter.java#L93

**Comment by gunnarmorling** [comment]
Hum, hum. Why is this even the case to begin with? I.e. why isn't that
value propagated as is?

**Comment by jpechane** [comment]
@gunnarmorling All zeroes are invalid date/time but allowed by MySQL as a sort of dummy date - see https://dev.mysql.com/doc/refman/8.0/en/date-and-time-types.html We were thinking about mapping it to `NULL` but the problem would be with `NOT NULL` columns. And as it is not a valid value it cannot be propagated as is.

**Comment by gunnarmorling** [comment]
@jpechane, ok turns out it already says this (added two years ago):

> MySQL allows zero-values for DATE, DATETIME, and TIMESTAMP columns, which are sometimes preferred over null values. These values cannot be represented using any of the Java types with either of the time.precision.mode options, and therefore the MySQL connector will represent them as null values when the column definition allows nulls, or as the epoch day when the column does not allow nulls.

So I think we're good? Also, has actually anything changed over that already described behavior with DBZ-191?

**Comment by jpechane** [comment]
@jpechane @sweat123 I am afraid we are not good. I 've missed this part and obviously we are now violating this contract. We use epoch always but we should in fact use epoch or null depending on constraints.

**Comment by sweat123** [comment]
@jpechane em. You are right! If the column definition allows null and the default value is all zero , we need set default value optional;

**Comment by gunnarmorling** [comment]
@jpechane So I think we can merge this PR now as is; with @sweat123's latest code change it should match what's described in the docs again?

**Comment by jpechane** [comment]
@gunnarmorling right

**Comment by gunnarmorling** [comment]
Excellent; merged into upstream/0.8.

**Pull request closed without merging**